### PR TITLE
[feature] (expr)  support expression IS [NOT] TURE/FALSE

### PR DESCRIPTION
**Pull request by justfortaste**
support expression IS [NOT] TURE/FALSE

```
mysql> select 1 is true, 1 is not true, 0 is true, 0 is not true;
+--------------------+---------------------------+--------------------+---------------------------+
| cast(1 as BOOLEAN) | ( not cast(1 as BOOLEAN)) | cast(0 as BOOLEAN) | ( not cast(0 as BOOLEAN)) |
+--------------------+---------------------------+--------------------+---------------------------+
|                  1 |                         0 |                  0 |                         1 |
+--------------------+---------------------------+--------------------+---------------------------+
```

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/backup/Untitled-2.cs [hunk 1]
```diff
@@ -0,0 +1,254 @@
++--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/backup/Untitled-2.cs:1]: what's this?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java [hunk 2]
```diff
@@ -3338,6 +3338,18 @@ private Expression withPredicate(Expression valueExpression, PredicateContext ct
                 case DorisParser.NULL:
                     outExpression = new IsNull(valueExpression);
                     break;
+                case DorisParser.TRUE:
+                    outExpression = new EqualTo(
+                            valueExpression,
+                            BooleanLiteral.of(1)
+                    );
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java:3345]: i think should cast left to boolean explicitily

File: regression-test/suites/nereids_syntax_p0/sql/test_compare_expression.sql [hunk 3]
```diff
@@ -34,3 +34,28 @@ select 1 not in (null, 2);
...
+select 1 in (2, null);
+
+select 1 not in (1, null);
+select 1 not in (2, null);
+
+select 1 in (2, null, 1);
+select 1 in (null, 2);
+
+select 1 not in (null, 1);
+select 1 not in (null, 2);
```
Comment (morrySnow) [inline regression-test/suites/nereids_syntax_p0/sql/test_compare_expression.sql:55]: add all type with `is (not) true | false`

File: regression-test/suites/nereids_syntax_p0/sql/test_compare_expression.sql [hunk 4]
```diff
@@ -34,3 +34,28 @@ select 1 not in (null, 2);
 select timestamp '2008-08-08 00:00:00' in ('2008-08-08');
 select case when true then timestamp '2008-08-08 00:00:00' else '2008-08-08' end;
 
+select 1 is true, 1 is not true, 0 is true, 0 is not true;
+
+
+
+
+
+SET enable_nereids_planner=false;
```
Comment (morrySnow) [inline regression-test/suites/nereids_syntax_p0/sql/test_compare_expression.sql:43]: do not test legacy planner explicitly, because we will remove it soon

**Comment by nextdreamblue** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by 924060929 — APPROVED** [review]

**Comment by 924060929** [comment]
run buildall

**Pull request merged**